src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -240,6 +240,32 @@ internal MySqlSession Session
 		internal bool AllowUserVariables => m_connectionStringBuilder.AllowUserVariables;
 		internal bool ConvertZeroDateTime => m_connectionStringBuilder.ConvertZeroDateTime;
 		internal bool OldGuids => m_connectionStringBuilder.OldGuids;
+		internal bool Synchronous => m_connectionStringBuilder.Synchronous;
+
+		internal Task AdaptTask(Task task)
+		{
+			if (!Synchronous)
+				return task;
+
+			task.GetAwaiter().GetResult();
+			return Task.FromResult<object>(null);
+		}
+
+		internal Task<T> AdaptTask<T>(Task<T> task)
+		{
+			if (!Synchronous)
+				return task;
+
+			return Task.FromResult(task.GetAwaiter().GetResult());
+		}
+
+		internal ValueTask<T> AdaptTask<T>(ValueTask<T> task)
+		{
+			if (!Synchronous)
+				return task;
+
+			return new ValueTask<T>(task.AsTask().GetAwaiter().GetResult());
```

Comment:
Should this have an optimised code path for the case where the `ValueTask` already has a value (which is expected to be common, hence the use of `ValueTask`)?